datastore/longhorn.go

```diff
@@ -2395,24 +2395,25 @@ func (s *DataStore) IsNodeDownOrDeletedOrMissingManager(name string) (bool, erro
 
 // IsNodeDownOrDeleted gets Node for the given name and namespace and checks
 // if the Node condition is gone or not ready
-func (s *DataStore) IsNodeDownOrDeleted(name string) (bool, error) {
+func (s *DataStore) IsNodeDownOrDeleted(name string) (nodeDelete bool, nodeDown bool, err error) {
```

Comment:
No need to modify this method, because what you want to determine is to check if node is deleted or not.

Create another method `isNodeDeleted` as below, and use it in the right place. Of course, remember to add doc comment to this exported method. 

```go
func (s *DataStore) IsNodeDeleted(name string) (bool, error) {
    if name == "" {
        return false, false, errors.New("no node name provided to check node down or deleted")
    }

    node, err := s.GetNodeRO(name)
    if err != nil {
        if apierrors.IsNotFound(err) {
            return true, nil
	}
        return false, err
    }

    cond := types.GetCondition(node.Status.Conditions, longhorn.NodeConditionTypeReady)

    return cond.Status == longhorn.ConditionStatusFalse && cond.Reason == longhorn.NodeConditionReasonKubernetesNodeGone, nil
}    
```